bin/processGsAndCopyRawDataToPrm.sh

```diff
@@ -0,0 +1,652 @@
+			for (( _offset = 0 ; _offset < ${#_sampleSheetColumnNames[@]:-0} ; _offset++ ))
+			do
+				_sampleSheetColumnOffsets["${_sampleSheetColumnNames[${_offset}]}"]="${_offset}"
+			done
+
+			declare -a _sequencingStartDate=()
+			local      _sequencingStartDateFieldIndex
+			if [[ ! -z "${_sampleSheetColumnOffsets['sequencingStartDate']+isset}" ]]; then
+			_sequencingStartDateFieldIndex=$((${_sampleSheetColumnOffsets['sequencingStartDate']} + 1))
+			IFS=$'\n' _sequencingStartDate=($(tail -n +2 "${_sampleSheet}" | cut -d "${SAMPLESHEET_SEP}" -f "${_sequencingStartDateFieldIndex}" | sort | uniq ))
+			if [[ "${#_sequencingStartDate[@]:-0}" -lt '1' ]]
+			then
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "${_sampleSheet} does not contain at least one project value."
+				log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Skipping ${_run} due to error in samplesheet."
+				touch "${_controlFileBase}.failed"
+				return
+			elif [[  "${#_sequencingStartDate[@]:-0}" -eq '1' ]]
+			then
+				echo "$_sequencingStartDate" >> "${_controlFileBase}.sequencingStartDate"
+			fi
+		else
+			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "project column missing in sample sheet."
+			log4Bash 'ERROR' "${LINENO}" "${FUNCNAME:-main}" '0' "Skipping ${_run} due to error in sample sheet."
+			touch "${_controlFileBase}.failed"
+			return
+		fi
+		done
+		touch "${_controlFileBase}.sanityChecking.finished"
+	else 
+		log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "_checksumVerification = ${_checksumVerification}"
+		echo "_checksumVerification = ${_checksumVerification} for run ${_run}" > "${_controlFileBase}.failed"
+		return
+	fi
+
+
+}
+
+function renameFastQs() {
+	local _run="${1}"
+	local _controlFileBase="${TMP_ROOT_DIR}/logs/${_run}/${_run}.${SCRIPT_NAME}"
+	local _logFile="${_controlFileBase}.log"
+	local _runPrefix="${TMP_ROOT_DIR}/${_run}/"
+
+if [[ -e "${_controlFileBase}.renameFastQs.finished" ]]
+then
+	#
+	# If previous sanityChecking was ok, skip
+	# 
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "${_controlFileBase}.renameFastQs.finished is there, nothing to do here."
+	return
+else
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${_controlFileBase}.renameFastQs.finished not present. continue"
+fi
+
+if [[ -e "${_controlFileBase}.sanityChecking.finished" && "${_controlFileBase}.sequencingStartDate" ]]
+then
+	#
+	# If previous sanityChecking was ok, skip
+	# 
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "${_controlFileBase}.sanityChecking.finished is there, ready for samplename renaming."
+	IFS=$'\n' _sequencingStartDate=($( cat "${_controlFileBase}.sequencingStartDate" | sort | uniq ))
+else
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${_controlFileBase}.sanityChecking.finished not present. return..."
+	return
+fi
+
+echo "_sequencingStartDate: ${_sequencingStartDate}"
+
+cd "${_runPrefix}"
+local _firstFastQ=$(ls -1 *.fastq.gz | head -1)
```

Comment:
No need to cd in this script to get list of fast.gz file; can be moved to sub shell to prevent surprises here.
Either something like:
```local _firstFastQ=$(cd "${_runPrefix}" && ls -1 *.fastq.gz | head -1)```
or
```local _firstFastQ=$(ls -1 "${_runPrefix}"/*.fastq.gz | head -1)```
should work and then you do not need to rely on paths relative to the current working directory of this script.